src/main/kotlin/com/github/adrienpessu/sarifviewer/toolWindow/SarifViewerWindowFactory.kt

```diff
@@ -380,14 +386,21 @@ class SarifViewerWindowFactory : ToolWindowFactory {
             treeBuilding(map)
         }
 
+        fun toggleRuleIdOrDescription() {
+            if (localMode) {
+                currentView.ruleIdToggle = !currentView.ruleIdToggle
+                extractSarifFromFile = extractSarifFromFile(selectedFile)
+                treeBuilding(extractSarifFromFile)
+            }
+        }
         fun openLocalFile() {
             val fileChooser = JFileChooser()
             fileChooser.fileFilter = FileNameExtensionFilter("SARIF files", "sarif")
             SwingUtilities.invokeLater {
                 val returnValue = fileChooser.showOpenDialog(null)
                 if (returnValue == JFileChooser.APPROVE_OPTION) {
                     val selectedFile: File = fileChooser.selectedFile
```

Comment:
```suggestion
                    selectedFile = fileChooser.selectedFile
```
without that the selectedFile attribute is never initiated and the toggle doesn't refresh the tree